gax-java/gax-grpc/src/main/java/com/google/api/gax/grpc/InstantiatingGrpcChannelProvider.java

```diff
@@ -410,6 +440,66 @@ ChannelCredentials createMtlsChannelCredentials() throws IOException, GeneralSec
     return null;
   }
 
+  @VisibleForTesting
+  ChannelCredentials createMtlsToS2AChannelCredentials(
+      InputStream trustBundle, InputStream privateKey, InputStream certChain) throws IOException {
+    if (trustBundle == null || privateKey == null || certChain == null) {
+      return null;
+    }
+    return TlsChannelCredentials.newBuilder()
+        .keyManager(privateKey, certChain)
+        .trustManager(trustBundle)
+        .build();
+  }
+
+  ChannelCredentials createS2ASecuredChannelCredentials() {
+    S2A s2aUtils = S2A.newBuilder().build();
+    String plaintextAddress = s2aUtils.getPlaintextS2AAddress();
+    String mtlsAddress = s2aUtils.getMtlsS2AAddress();
+    if (!mtlsAddress.isEmpty()) {
```

Comment:
Can we change this to:
```
if (mtlsAddress.isEmpty()) {
  return null;
}
```
to reduce nesting?

Also don't we need this for plaintextAddress as well?

Reply:
The logic we want is to "fallback" to plaintextAddress in a few cases (basically on any failure to create mtls to S2A creds):
- `mtlsAddress.isEmpty()`
- if MTLS-MDS files are not found
- error creating `TlsChannelCredentials`

In https://github.com/googleapis/sdk-platform-java/pull/3326/commits/30a37c221432cbf312ccdc8728dad5da23cf316e I reduced the nesting, let me know what you think. I also reversed the negation checks (e.g. `if x != null) to make it a bit more readable.